unit/testing-utils/require_type.cpp

```diff
@@ -78,3 +93,105 @@ code_typet::parametert require_type::require_parameter(
   REQUIRE(param != function_type.parameters().end());
   return *param;
 }
+
+/// Verify a given type is a java_generic_type, optionally checking
+/// that it's associated type variables match a given set of identifiers
+/// \param type The type to check
+/// \param type_variables An optional set of type variable identifiers which
+/// should be expected as the type parameters of the generic type.
+/// \return The given type, cast to a java_generic_typet
+const java_generic_typet &require_type::require_java_generic_type_variables(
+  const typet &type,
+  const optionalt<std::initializer_list<irep_idt>> &type_variables)
```

Comment:
Since we will have generic types with both instantiated and uninstantiated parameters, this may not need to be a complete list of parameters.